engine/engine.go

```diff
@@ -73,6 +73,7 @@ func (engine *Engine) Loop() {
 				engine.State.ActiveRooms += 1
 			} else {
 				engine.State.ClosedRooms += 1
+				delete(engine.rooms.m, rid)
```

Comment:
In line 52, engine.rooms.RLock() should be changed to engine.rooms.Lock(). Line 80 also needs change.